### ISSUE TITLE
Automate spell checking

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#2504]
## Feature Description

Currently we have no form of automated test or tooling in place for checking the spelling of user-facing strings. While uncommon, there have been a few spelling errors which have slipped through (#2401).

This would also be useful in catching potential English-specific misspellings (e.g. using en-GB instead of en-US).

As an open-source project with contributors and maintainers from many different countries, it would be useful to leverage some form of automated spell checking using GitHub actions and/or ESLint, etc.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
@felixarntz check out https://github.com/marketplace/typo-ci

This would be great to add ahead of our upcoming work to add contributor docs.

**Comment by aaemnnosttv** [comment]
@felixarntz anything to add?

**Comment by felixarntz** [comment]
@aaemnnosttv No, all good from my perspective! 🚢

**Comment by danielgent** [comment]
![Screenshot from 2021-05-06 11-08-25](https://user-images.githubusercontent.com/8496063/117272361-697d2c00-ae5b-11eb-978d-8ab7064cd858.png)

Is the plan to set up a free trial and see how it goes?  (from the description on the ticket I was not expecting a paid product :smile: )

**Comment by aaemnnosttv** [comment]
> Is the plan to set up a free trial and see how it goes?

@danielgent nope! It's a free service for open source (the first option in the list there) 🙂

**Comment by danielgent** [comment]
I've created a PR with different kinds of typos if you want to see it in action https://github.com/google/site-kit-wp/pull/3324

Works really well

**Comment by aaemnnosttv** [comment]
Thanks @danielgent – IB looks good, just a few comments/suggestions:

> matching patterns for source files that may contain user-facing strings

We don't need to list specific paths here but it would be nice to note/outline which file types or locations we should base the `paths` on .

We may also want to scaffold the `excluded_words` entry to the config, maybe with a `googlesitekit` in there – we should check to see if it flags this and the all-caps variant (we use both). Does it see all lower/upper the same? I would think it probably does but that would be good to know.

I was going to suggest we only run it for non-draft PRs but then I saw it only took 4 seconds to run, so I think it's fine 😄  Also, I think it only checks text in the diff rather than the entire code base which is what we'd want anyways.

Let's check on the case for any common things which might be flagged as misspellings which aren't in our codebase (like `googlesitekit`, `GOOGLESITEKIT`, `tagmanager`, `webdatastreams`, etc).

**Comment by danielgent** [comment]
Typo CI will be shut down soon https://mikerogers.io/2021/05/23/saying-goodbye-to-typo-ci

**Comment by aaemnnosttv** [comment]
Moving this to _stalled_ for now.

**Comment by aaemnnosttv** [comment]
Thanks @zutigrm – I am still interested in moving this forward if there is a good option. I took a look at Vale and that isn't going to work since this issue is about spellchecking across the whole codebase (e.g. code comments, variable names, etc), whereas Vale is specific to written content like documentation and so it's geared towards spellchecking various forms of markup 

<img width="402" height="361" alt="Image" src="https://github.com/user-attachments/assets/9bb2eb86-5e5a-4eda-b1a6-58184e4cc1c1" />
